### PR TITLE
reword some assertions with more informations

### PR DESCRIPTION
**Pull request by romankl**
## Changes in this pull request
This commit changes some assertion texts. No behavior is changed; no assertions are added (just reworded).  Some of them were short or contained no real meaningful informations, where the error probably is.

Ref: #113

## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: Source/IGListAdapter.m [hunk 1]
```diff
@@ -716,7 +716,7 @@ - (__kindof UICollectionViewCell *)dequeueReusableCellOfClass:(Class)cellClass
     IGParameterAssert(sectionController != nil);
     IGParameterAssert(cellClass != nil);
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Reloading adapter without a collection view.");
+    IGAssert(collectionView != nil, @"Tried to reload the adapter from %@ without a collection view at index %zi.", sectionController, index);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: Can this say something like `Dequeueing cell of class %@ without a collection view at section %zi`?

File: Source/IGListAdapter.m [hunk 2]
```diff
@@ -733,7 +733,7 @@ - (UICollectionViewCell *)dequeueReusableCellWithNibName:(NSString *)nibName
     IGAssertMainThread();
     IGParameterAssert(sectionController != nil);
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Reloading adapter without a collection view.");
+    IGAssert(collectionView != nil, @"Tried to reload the adapter from %@ without a collection view at index %zi.", sectionController, index);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: Same note

File: Source/IGListAdapter.m [hunk 3]
```diff
@@ -749,7 +749,7 @@ - (__kindof UICollectionReusableView *)dequeueReusableSupplementaryViewOfKind:(N
                                                                       atIndex:(NSInteger)index {
     IGAssertMainThread();
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Reloading adapter without a collection view.");
+    IGAssert(collectionView != nil, @"Tried to reload the adapter from %@ without a collection view at index %zi.", sectionController, index);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: Same note but mention the `elementKind`

File: Source/IGListAdapter.m [hunk 4]
```diff
@@ -764,7 +764,7 @@ - (void)reloadInSectionController:(IGListSectionController<IGListSectionType> *)
     IGParameterAssert(indexes != nil);
     IGParameterAssert(sectionController != nil);
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Reloading from %@ without a collection view.", sectionController);
+    IGAssert(collectionView != nil, @"Tried to reload the adapter from %@ without a collection view at index %@.", sectionController, indexes);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: Can we also use the `indexes` param?

File: Source/IGListAdapter.m [hunk 5]
```diff
@@ -779,7 +779,7 @@ - (void)insertInSectionController:(IGListSectionController<IGListSectionType> *)
     IGParameterAssert(indexes != nil);
     IGParameterAssert(sectionController != nil);
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Inserting items from %@ without a collection view.", sectionController);
+    IGAssert(collectionView != nil, @"Inserting items from %@ without a collection view at index %@.", sectionController, indexes);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: nit: "indexes"

File: Source/IGListAdapter.m [hunk 6]
```diff
@@ -794,7 +794,7 @@ - (void)deleteInSectionController:(IGListSectionController<IGListSectionType> *)
     IGParameterAssert(indexes != nil);
     IGParameterAssert(sectionController != nil);
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Deleting items from %@ without a collection view.", sectionController);
+    IGAssert(collectionView != nil, @"Deleting items from %@ without a collection view at index %@.", sectionController, indexes);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: "indexes"

**Comment by romankl** [comment]
@rnystrom thanks for the review! I've updated the PR with the requested changes

File: Source/IGListAdapter.m [hunk 7]
```diff
@@ -749,7 +749,7 @@ - (__kindof UICollectionReusableView *)dequeueReusableSupplementaryViewOfKind:(N
                                                                       atIndex:(NSInteger)index {
     IGAssertMainThread();
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Reloading adapter without a collection view.");
+    IGAssert(collectionView != nil, @"Dequeueing cell of class %@ without a collection view at section %zi with supplementary view %@", sectionController, index, elementKind);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: "cell of class" param should be the `viewClass`

File: Source/IGListAdapter.m [hunk 8]
```diff
@@ -764,7 +764,7 @@ - (void)reloadInSectionController:(IGListSectionController<IGListSectionType> *)
     IGParameterAssert(indexes != nil);
     IGParameterAssert(sectionController != nil);
     UICollectionView *collectionView = self.collectionView;
-    IGAssert(collectionView != nil, @"Reloading from %@ without a collection view.", sectionController);
+    IGAssert(collectionView != nil, @"Tried to reload the adapter from %@ without a collection view at indexes %@.", sectionController, indexes);
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: ❤️

**Comment by romankl** [comment]
@rnystrom thanks again for the review! I've pushed a new commit that includes your changes

**Comment by rnystrom** [comment]
@romankl woot! Imported. Waiting on travis and internal CI to be done and green, then landing.

**Pull request closed without merging**